### PR TITLE
Add support for generating unique DuckDuckGo Email Protection addresses

### PR DESCRIPTION
**Pull request by Rediwed**
# Description

DuckDuckGo Email Protection is an email protection service (currently invite-only). After having been invited, DDG provides you with an intermediary e-mail address ([you@duck.com](mailto:you@duck.com)). Any e-mail sent to this address will be stripped of its analytics components, before being forwarded to your actual e-mail address ([you@you.com](mailto:you@you.com)).

This script command lets you quickly generate a unique @duck.com address that is connected to your own e-mail address. Just like your personal [you@duck.com](mailto:you@duck.com) address, the unique addresses will also forward any e-mail to your own e-mail address, after having been stripped of any tracking or analytics modules.

# Type of change

- [x] New Script Command

(I have added this script command to the communication group, becuase it is related to generating unique email addresses instead of related to performing web-searches)

# Screenshot

Screenshot of Microsoft Excel on 2022-02-08 at 14 24 07

Dependencies / Requirements

- It is required to have been invited into the @duck.com email protection service.
- It is also required to extract your authorization ID from the official DuckDuckGo Privacy Essentials extension (see README.md).

# Checklist

- [x] I have read [Contribution Guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: commands/communication/duckduckgo-email-protection/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+# What is DuckDuckGo Email Protection?
+DuckDuckGo Email Protection is an email protection service (currently invite-only). After having been invited, DDG provides you with an intermediary e-mail address (you@duck.com). Any e-mail sent to this address will be stripped of its analytics components, before being forwarded to your actual e-mail address (you@you.com).
+
+![image](https://user-images.githubusercontent.com/9107063/152996647-676ff3df-4ebe-49a5-a28d-97d2fe21ec94.png)
```
Comment (dehesa) [inline commands/communication/duckduckgo-email-protection/README.md:5]: We don't currently accept images to external services. Could you add it to the PR under the `images` folder.
Be sure to reduce the pixel density as much as you can (while still being readable) and pass the image by [ImageOptim](https://imageoptim.com/mac) (or similar compressors).

File: commands/communication/duckduckgo-email-protection/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+# About this Script Command
+This script command lets you quickly generate a unique @duck.com address that is connected to your own e-mail address. Just like your personal you@duck.com address, the unique addresses will also forward any e-mail to your own e-mail address, after having been stripped of any tracking or analytics modules.
+
+![Screenshot of Raycast on 2022-02-08 at 14 24 07](https://user-images.githubusercontent.com/9107063/152995951-91c9f7bd-eddd-43a3-98b0-79dad33308eb.png)
```
Comment (dehesa) [inline commands/communication/duckduckgo-email-protection/README.md:11]: Same thing here.

File: commands/communication/duckduckgo-email-protection/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+2. Then, navigate go to about:debugging#/runtime/this-firefox
+3. In the list of installed extension, look for DuckDuckGo Privacy Essentials and click 'Inspect'
+4. In the inspector, navigate to the Network tab
+5. Click on the extension icon, and click on 'Create new Duck Address' (bottom)
+6. A new POST request will have appeared in the network tab, select it
+7. In the 'Request Headers' section, look for 'Authorization'
+8. Copy the string of randomized characters AFTER "Bearer "
+9. Open Raycast and paste the string in the Configure script command. You will receive a confirmation should the string be correct.
+
+![Screenshot of Firefox on 2022-02-08 at 14 21 47](https://user-images.githubusercontent.com/9107063/152995838-fe619525-b4dd-4d3d-bf7f-95c07ed7b2d2.png)
```
Comment (dehesa) [inline commands/communication/duckduckgo-email-protection/README.md:32]: and here.

File: commands/communication/duckduckgo-email-protection/generate-unique-email-address.applescript [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/osascript
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Generate unique address
```
Comment (dehesa) [inline commands/communication/duckduckgo-email-protection/generate-unique-email-address.applescript:5]: It should be titlecased; i.e. `Generate Unique Address`

**Review by dehesa — CHANGES_REQUESTED** [review]
Hi @Rediwed,

This looks pretty useful. I wasn't aware of it. Cool stuff! I left you some comments below.

By the way, these scripts would benefit from being an [extension](https://developers.raycast.com/). Check them out, if you weren't aware of it.

File: commands/communication/duckduckgo-email-protection/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+2. Then, navigate go to about:debugging#/runtime/this-firefox
+3. In the list of installed extension, look for DuckDuckGo Privacy Essentials and click 'Inspect'
+4. In the inspector, navigate to the Network tab
+5. Click on the extension icon, and click on 'Create new Duck Address' (bottom)
+6. A new POST request will have appeared in the network tab, select it
+7. In the 'Request Headers' section, look for 'Authorization'
+8. Copy the string of randomized characters AFTER "Bearer "
+9. Open Raycast and paste the string in the Configure script command. You will receive a confirmation should the string be correct.
+
+![Locate authorization](locate-authorization-id.png)
```
Comment (dehesa) [inline commands/communication/duckduckgo-email-protection/README.md:32]: You forgot the `images/` prefix 😄

**Review by dehesa — CHANGES_REQUESTED** [review]

File: commands/communication/duckduckgo-email-protection/README.md (hunk 5 above)
Comment (Rediwed) [reply commands/communication/duckduckgo-email-protection/README.md]: Darn it! Fixed it now :)

**Comment by dehesa** [comment]
Thanks for the great contribution, @Rediwed

**Comment by Rediwed** [comment]
> By the way, these scripts would benefit from being an [extension](https://developers.raycast.com/). Check them out, if you weren't aware of it.

Sorry, overread this comment. Thank you for pointing it out, though I was aware of Extensions.

I'm not comfortable with React, Node.js, or TypeScript. Otherwise, I definitely would've made an extension 🙂 I've worked with AppleScript (and AppleScriptObjC) for the more than a decade, and recently started working with Python (though mostly for data-science, not really programming).

What made me so much more excited for Raycast (over Alfred or Launchbar) was that I could actually augment it with AppleScript. So thank you so much for allowing that!

**Comment by dehesa** [comment]
I understand. Many thanks for taking the time to contribute this to the community.

**Pull request merged**